samples/nextjs/src/lib/page-props-factory.ts

```diff
@@ -0,0 +1,149 @@
+import { AxiosError } from 'axios';
+import { ParsedUrlQuery } from 'querystring';
+import { GetServerSidePropsContext, GetStaticPropsContext, NextPageContext } from 'next';
+import {
+  ComponentPropsService,
+  EditingRequest,
+  LayoutService,
+  LayoutServiceData,
+} from '@sitecore-jss/sitecore-jss-nextjs';
+import { SitecorePageProps } from 'lib/page-props';
+import { componentModule } from 'temp/componentFactory';
+import { layoutService } from 'lib/layout-service';
+import { DictionaryService, dictionaryService } from 'lib/dictionary-service';
+import { config as packageConfig } from '../../package.json';
+
+/**
+ * Extract normalized Sitecore item path from query
+ */
+const extractPath = function (params: ParsedUrlQuery | undefined): string {
+  if (params === undefined) {
+    return '/';
+  }
+  let path = Array.isArray(params.path) ? params.path.join('/') : params.path ?? '/';
+
+  // Ensure leading '/'
+  if (!path.startsWith('/')) {
+    path = '/' + path;
+  }
+
+  return path;
+};
+
+/**
+ * Determine if Next.js context is GetServerSidePropsContext
+ * @param context {GetStaticPropsContext|GetServerSidePropsContext}
+ */
+const isGetServerSidePropsContext = function (
+  context: GetServerSidePropsContext | GetStaticPropsContext
+): context is GetServerSidePropsContext {
+  return 'req' in context;
+};
+
+export class SitecorePagePropsFactory {
+  private get componentPropsService(): ComponentPropsService {
+    return new ComponentPropsService();
+  }
+
+  private get layoutService(): LayoutService {
+    // Just returning our REST layout service atm, but in the very
+    // near future we'll also have a GraphQL-based layout service.
+    // Stubbed out as getter for potential logic here (e.g. based on constructor props)...
+    return layoutService;
+  }
+
+  private get dictionaryService(): DictionaryService {
+    // Just returning our REST dictionary service atm, but in the very
+    // near future we'll also have a GraphQL-based dictionary service.
+    // Stubbed out as getter for potential logic here (e.g. based on constructor props)...
+    return dictionaryService;
+  }
+
+  /**
+   * Create SitecorePageProps in an SSG/SSR context (within getStaticProps/getServerSideProps function).
+   * @param context {GetStaticPropsContext|GetServerSidePropsContext}
+   * @see SitecorePageProps
+   */
+  public async create<T extends GetStaticPropsContext | GetServerSidePropsContext>(
+    context: T
+  ): Promise<SitecorePageProps> {
+    // Get normalized Sitecore item path
+    const path = extractPath(context.params);
+    // Use context locale if Next.js i18n is configured, otherwise use language defined in package.json
+    const locale = context.locale ?? packageConfig.language;
+
+    let notFound = false;
+
+    // Retrieve layoutData from Layout Service
+    const layoutData = await this.layoutService
+      .fetchLayoutData(path, locale)
```

Comment:
@ambrauer We don't have an option to execute `fetchLayoutData` using `req`, `res`, as it was previously, so it's only working for SSG

Reply:
@sc-illiakovalenko Great catch, those got lost in the shuffle. Fixed.